android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java

```diff
@@ -349,30 +367,43 @@ public boolean setKeyboard(String packageID, String keyboardID, String languageI
     //  return false;
 
     keyboardVersion = KMManager.getLatestKeyboardFileVersion(getContext(), packageID, keyboardID);
+
+    String keyboardPath = "";
+    if (packageID.equals(KMManager.KMDefault_UndefinedPackageID)) {
```

Comment:
This filename and path construction needs to be extracted into a separate helper function because it is duplicated here and above.

Reply:
I'm moving those into helper functions `setKeyboardRoot()` and `makeKeyboardPath()`